src/tests/banners/bannerSelection.tsx

```diff
@@ -4,25 +4,36 @@ import {
     BannerTestSelection,
 } from '../../components/BannerTypes';
 import { AusMomentContributionsBanner } from './AusMomentContributionsBannerTest';
+import { SubscriptionsBanner } from './SubscriptionsBannerTest';
+import { WeeklyBanner } from './WeeklyBannerTest';
 import fetch from 'node-fetch';
 import { cacheAsync } from '../../lib/cache';
+import { countryCodeToCountryGroupId, CountryGroupId } from '../../lib/geolocation';
 
-type ReaderRevenueRegion = 'united-kingdom' | 'united-states' | 'australia' | 'rest-of-world';
+type ReaderRevenueRegion =
+    | 'united-kingdom'
+    | 'united-states'
+    | 'australia'
+    | 'rest-of-world'
+    | 'european-union';
 
-const readerRevenueRegionFromCountryCode = (countryCode: string): ReaderRevenueRegion => {
-    switch (countryCode.toLowerCase()) {
-        case 'gb':
+export const readerRevenueRegionFromCountryCode = (countryCode: string): ReaderRevenueRegion => {
+    const countryGroupId: CountryGroupId = countryCodeToCountryGroupId(countryCode);
+    switch (countryGroupId) {
+        case 'GBPCountries':
             return 'united-kingdom';
-        case 'us':
+        case 'UnitedStates':
             return 'united-states';
-        case 'au':
+        case 'AUDCountries':
             return 'australia';
+        case 'EURCountries':
+            return 'european-union';
         default:
             return 'rest-of-world';
     }
 };
 
-const fetchBannerDeployTime = (region: ReaderRevenueRegion) => (): Promise<Date> => {
+const fetchContributionsBannerDeployTime = (region: ReaderRevenueRegion) => (): Promise<Date> => {
```

Comment:
I would probably do this:

```javascript
export const fetchBannerDeployTime = (region: ReaderRevenueRegion, bannerType: 'contributions' | 'subscriptions') => (): Promise<
    Date
> => {
    return fetch(
        `https://www.theguardian.com/reader-revenue/${bannerType}-banner-deploy-log/${
            region === 'european-union' ? 'united-kingdom' : region
        }`,
    )
        .then(response => response.json())
        .then(data => {
            return new Date(data.time);
        });
};

const fetchContributionsBannerDeployTime = (region: ReaderRevenueRegion) => (): Promise<Date> => 
  fetchBannerDeployTime(region, 'contributions');

const fetchSubscriptionsBannerDeployTime = (region: ReaderRevenueRegion) => (): Promise<Date> => 
  fetchBannerDeployTime(region, 'subscriptions');

Reply:
This is tidy, unfortunately we have the problem that the `region === 'european-union' ? 'united-kingdom'` is true for subscriptions, for contributions however `region === 'european-union' ? rest-of-world'` is true.

This is because no 'european-union'  redeploy button exists  yet so subscriptions have decided to group  redeploys to 'european-union' under the 'united-kingdom' redeploy button, and contributions have grouped them under the 'rest-of-world'.

Reply:
Ah yes, so it should be:
```javascript
export const fetchBannerDeployTime = (region: ReaderRevenueRegion, bannerType: 'contributions' | 'subscriptions') => (): Promise<
    Date
> => {
    return fetch(
        `https://www.theguardian.com/reader-revenue/${bannerType}-banner-deploy-log/${region}`,
    )
        .then(response => response.json())
        .then(data => {
            return new Date(data.time);
        });
};

const fetchContributionsBannerDeployTime = (region: ReaderRevenueRegion) => (): Promise<Date> => 
  fetchBannerDeployTime(region, 'contributions');

const fetchSubscriptionsBannerDeployTime = (region: ReaderRevenueRegion) => (): Promise<Date> => 
  fetchBannerDeployTime(region === 'european-union' ? 'united-kingdom' : region, 'subscriptions');
```